sdk/node/test/accounts.js

```diff
@@ -0,0 +1,246 @@
+/* eslint-env mocha */
+
+const chain = require('../dist/index.js')
+const uuid = require('uuid')
+const assert = require('assert')
+const chai = require('chai')
+const chaiAsPromised = require('chai-as-promised')
+
+chai.use(chaiAsPromised)
+const expect = chai.expect
+
+const client = new chain.Client()
+const signer = new chain.HsmSigner()
+
+const aliceAlias = `alice-${uuid.v4()}`
+const bobAlias = `bob-${uuid.v4()}`
+
+let aliceKey, bobKey, otherKey
+
+describe('Accounts test', () => {
+
+  describe('Promise style', () => {
+
+    before('set up API objects', () => {
+
+      // Key creation and signer setup
+      return expect(Promise.all([
+        client.mockHsm.keys.create({alias: aliceAlias}),
+        client.mockHsm.keys.create({alias: bobAlias}),
+        client.mockHsm.keys.create()
+      ])).to.be.fulfilled
+
+      .then(keys => {
+        aliceKey = keys[0]
+        bobKey = keys[1]
+        otherKey = keys[2]
+
+        signer.addKey(aliceKey, client.mockHsm.signerConnection)
+        signer.addKey(bobKey, client.mockHsm.signerConnection)
+      })
+    })
+
+    describe('Single account creation', () => {
+
+      it('creates two accounts', () => {
+        return expect(Promise.all([
+          client.accounts.create({alias: aliceAlias, rootXpubs: [aliceKey.xpub], quorum: 1}),
+          client.accounts.create({alias: bobAlias, rootXpubs: [bobKey.xpub], quorum: 1})
+        ])).to.be.fulfilled
+      })
+
+      it('account creation rejected due to missing key fields', () => {
+        return expect(client.accounts.create({alias: 'david'}))
+        .to.be.rejectedWith('CH202')
+      })
+    })
+
+    describe('Batch account creation', () => {
+      let batchResponse = {}
+
+      before(() => client.accounts.createBatch([
+        {alias: `carol-${uuid.v4()}`, rootXpubs: [otherKey.xpub], quorum: 1}, // success
+        {alias: 'david'}, // failure
+        {alias: `eve-${uuid.v4()}`, rootXpubs: [otherKey.xpub], quorum: 1}, // success
+      ])
+      .then(resp => {batchResponse = resp}))
+
+      it('returns two successes', () => assert.equal(batchResponse.successes[1], null))
+      it('returns one error', () => assert.deepEqual([batchResponse.errors[0], batchResponse.errors[2]], [null, null]))
+    })
+
+    describe('Single account tags update', () => {
+
+      it('successfully updates account tags', () => {
+        return Promise.resolve()
+
+        .then(() => {
+          return expect(
+            client.accounts.create({
```

Comment:
We could probably create two accounts in a before block here and remove that from single and batch account tag test cases.